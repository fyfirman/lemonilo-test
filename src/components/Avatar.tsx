import styled from 'styled-components/native';

interface AvatarProps {
  size?: number;
}

const DEFAULT_SIZE = 40;

const Avatar = styled.Image<AvatarProps>`
  border-radius: 100px;
  width: ${(props) => props.size ?? DEFAULT_SIZE}px;
  height: ${(props) => props.size ?? DEFAULT_SIZE}px;
`;

export default Avatar;
