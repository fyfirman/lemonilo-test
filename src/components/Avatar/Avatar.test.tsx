import React from 'react';
import { renderWithTheme } from '../../__tests__/helpers';
import Avatar from './Avatar';
import 'jest-styled-components';

const image = require('@assets/images/avataaars.png');

test('should renders correctly', () => {
  const tree = renderWithTheme(<Avatar source={image} />);

  expect(tree).toMatchSnapshot();
});
