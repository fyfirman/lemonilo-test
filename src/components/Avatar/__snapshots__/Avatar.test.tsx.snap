// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/assets/images/avataaars.png",
    }
  }
  style={
    Array [
      Object {
        "backgroundColor": "#19A0E0",
        "borderRadius": 100,
        "height": 40,
        "width": 40,
      },
    ]
  }
/>
`;
